db/db_impl_compaction_flush.cc

```diff
@@ -2170,6 +2170,10 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
       manual_compaction->in_progress = false;
       manual_compaction = nullptr;
     }
+    if (c) {
```

Comment:
NIT: In the description, you mentioned manual compaction, but here it seems that we need to release the files for any pre-picked compaction.

Reply:
Right, good point.

Reply:
clarified in the description but left the title since I believe manual compaction is the dominant use of pre-picked compaction.